client/src/features/users/api/register.ts

```diff
@@ -0,0 +1,30 @@
+import { useMutation } from "@tanstack/react-query";
+import { z } from "zod";
+import { api } from "../../../shared/config/api-client";
+import { MutationConfig } from "../../../shared/config/react-query";
+import { User } from "../types";
+
+export const registerInputSchema = z.object({
+  firstName: z.string().min(1, "First name is required."),
+  lastName: z.string().min(1, "Last name is required."),
+  phoneNumber: z.string().min(1, "Phone number is required."),
+  email: z.string().min(1, "Email is required."),
```

Comment:
```suggestion
  email: z.string().min(1, "Email is required.").email("Invalid email."),
```
maybe add zod check for email format?

Reply:
added